src/viser/client/src/CameraControls.tsx

```diff
@@ -113,11 +113,17 @@ export function SynchronizedCameraControls() {
     setTimeout(() => sendCamera(), 50);
   }, [connected, sendCamera]);
 
+  // Send camera for 3D viewport changes.
   React.useEffect(() => {
-    window.addEventListener("resize", sendCamera);
-    return () => {
-      window.removeEventListener("resize", sendCamera);
-    };
+    // Create a resize observer to resize the CSS canvas when the window is resized.
+    const resizeObserver = new ResizeObserver(() => { sendCamera() });
```

Comment:
also should we use the throttled camera sender? `sendCameraThrottled()`?

Reply:
Hmm, I kept it as `sendCamera` because that was what was originally used in the `useEffect` block. 

What is the main difference between `sendCamera` and `sendCameraThrottled`? Isn't `sendCamera` a wrapper?

Reply:
Oh sorry you're right! My naming here was just unclear (`sendCamera` is really more like `sendCameraThrottledWithPopulatedInputs`), your implementation looks good.